src/vulkan-renderer/texture.cpp

```diff
@@ -1,27 +1,331 @@
 #include "inexor/vulkan-renderer/texture.hpp"
 
+// stb single-file public domain libraries for C/C++
+// https://github.com/nothings/stb
+// License: Public Domain
+#define STB_IMAGE_IMPLEMENTATION
+#include <stb_image.h>
+
 namespace inexor::vulkan_renderer {
 
-void Texture::destroy_texture(const VkDevice &device, const VmaAllocator &vma_allocator) {
-    vkDestroySampler(device, sampler, nullptr);
+Texture::Texture(Texture &&other) noexcept
+    : name(std::move(other.name)), file_name(std::move(other.file_name)), texture_width(other.texture_width), texture_height(other.texture_height),
+      texture_channels(other.texture_channels), mip_levels(other.mip_levels), device(other.device), graphics_card(other.graphics_card),
+      data_transfer_queue(other.data_transfer_queue), vma_allocator(other.vma_allocator), allocation(std::exchange(other.allocation, nullptr)),
+      allocation_info(other.allocation_info), image(std::exchange(other.image, nullptr)), image_view(std::exchange(other.image_view, nullptr)),
+      sampler(std::exchange(other.sampler, nullptr)), texture_image_format(other.texture_image_format),
+      copy_command_buffer(std::move(other.copy_command_buffer)) {}
 
-    vmaDestroyImage(vma_allocator, image, allocation);
+// TODO: Remove unnecessary parameters!
+Texture::Texture(const VkDevice device, const VkPhysicalDevice graphics_card, const VmaAllocator vma_allocator, void *texture_data,
```

Comment:
If not already mentioned in the header. Reuse the one constructor as both share a lot of code.

Reply:
That's a difficult thing.

The problem is that the other constructor would have to be called after `stbi_load` is done.
Since that function is call by value for `texture_width`, `texture_height`, and `texture_channels`, I can't just call the other constructor.

I could however move the code to a private member function which is called by both constructors.

Reply:
Ok I will create a private member function.

Reply:
done.